app/helpers/spree/admin/slides_helper.rb

```diff
@@ -14,6 +14,20 @@ def get_image_link_by_type(slide, type)
 
         return '----'
       end
+
+      def product_options_for_select(slide)
+        if slide.product_id.present?
+          Spree::Product.where(id: slide.product_id).to_a
+        else
+          []
+        end
+
+      end
+
+      def product_url(slug)
```

Comment:
we should not require to add this helper..

rails routes based helper are already enough for this..and more reliable

Reply:
we do have 
```
include Rails.application.routes.url_helpers
```

but it doesn't help us to get generate product_path `/products/:id`